ios/ExpoNotificationServiceExtension/NotificationService.swift

```diff
@@ -0,0 +1,39 @@
+/*
+ * This class allows you to intercept and mutate incoming remote notifications.
+ * didReceive() has ~30 seconds to modify the payload and call the contentHandler,
+ * otherwise serviceExtensionTimeWillExpire() will be triggered.
+ *
+ * The notification payload must contain:
+ *    "mutable-content" : 1
+ *    "alert" : { ... }
```

Comment:
Yeah I agree, but it's a [requirement from Apple](https://developer.apple.com/documentation/usernotifications/modifying_content_in_newly_delivered_notifications), otherwise the extension just won't process the notification:

> The system executes your notification service app extension only when a remote notification’s payload contains the following information:
> - The payload must include the mutable-content key with a value of 1.
> - The payload must include an alert dictionary with title, subtitle, or body information.

Reply:
We can conditionally send the mutable-content field since we know if an client is running Expo Go, but we don't control whether notifications contain a title/subtitle/body -- developers might send notifications that only set a badge counter or play a sound.

Just as a sanity check: are there any scenarios in iOS programming in which a notification doesn't show an alert but a category would have an effect? Based on the API docs, it looks like categories on iOS are used for:

- custom action buttons (need an alert)
- associating notifications with intents, namely from Siri (this I'm not sure about...)
- controlling how notifications are displayed on the lock screen (applies only to alerts)
- controlling how notifications are grouped (applies only to alerts)

I think we could move ahead with this approach but it would mean that we won't be able to send alertless notifications with a category to Expo Go. In practice this seems okay for now since categories are used for visual notifications and we don't support the Intent API in Expo Go nor is it clear what happens when a notification is associated with an intent.

Reply:
What do you think of _defaulting_ to `mutableContent: true` for Expo Go, **but** still allowing devs to override it by providing `mutableContent: false` in their own payload? And say "this is what _won't_ happen if you override this, so it's up to you to modify your notification content based on that"

To me, it doesn't look like iOS _intends_ for notification categories to be used with alertless notifications (since they describe categories as "...objects to define your app's actionable notifications—that is, notifications that have action buttons that the user can select in response to the notification.") but I'm sure someone somewhere has used it for that case.

Reply:
Switching the default could be a good idea so people can still opt out if needed.